src/Parsers/ExpressionListParsers.cpp

```diff
@@ -603,6 +605,13 @@ bool ParserTableFunctionExpression::parseImpl(Pos & pos, ASTPtr & node, Expected
 {
     if (ParserTableFunctionView().parse(pos, node, expected))
         return true;
+    ParserKeyword s_settings("SETTINGS");
+    if (s_settings.ignore(pos, expected))
```

Comment:
I am not sure if it was a right way to put this into all table functions parsing and not only for executable table function. I have I special parameters format in one table function which is in development and now I cannot use `settings` keyward in its arguments and cannot use ASTSetQuery as it is not consistent with parsing of other arguments. :(

Reply:
It is generic way of adding settings to any table function, otherwise each function can add settings in its own way, we already have examples when there are 5-9 function arguments, most of them are just unnamed settings.
Parser should not know anything about any table function, if we put if function = 'executable' and parse settings only for it, it will be huge problem later. 
If your table function need some special 'settings' you can prefix 'settings' name with it.

Reply:
Ok, also makes sense.

> If we put if function = 'executable' and parse settings only for it, it will be huge problem later.

I did not mean it. Settings could be extracted based on settings keyword not on parser level but in `parseArguments` method of table function.

Reply:
I am not sure if such construction `table_function(..., SETTINGS max_command_execution_time=100, command_read_timeout = 100000)`  was allowed on parser level before (probably not and we added it in this pull request).

Construction like `table_function(..., settings(max_command_execution_time=100, command_read_timeout = 100000))`, `table_function(..., executable_settings(max_command_execution_time=100, command_read_timeout = 100000))` were not approved by @alexey-milovidov. Reasoning was that user already expects `SETTINGS` to be used for settings, not some ad-hoc constructions, and also additional benefit that current approach makes syntax validation on parser stage.

Also feel free to change it to allow settings function and also SETTINGS keyword on parser level, you can check if client mean SETTINGS or `settings` function, if it is important for you task, but if you can reuse SETTINGS it will be more consistent.

Reply:
The parser was changed to support any table function with a `SETTING key = value,...` list.
So it probably should be reused.